### PR TITLE
Auth backend set up + testing

### PR DESCRIPTION
**Pull request by colton456p**
## PR Info
- Set up **passport.js** for our backend authentication along with **bcrypt** for our password hashing, and **express.js** for our session management and routing. I wrote backend routes login, signup, logout and the user info return. 
- Set up our test environment using **Mocha**, **Chai**, and **Supertest** and set up a project linter on our `tests` folder.
- Set up automated testing with git hub actions. This runs docker to get the DB up and running, then runs a test to insure there is no issues with the db, then runs all the backend tests insuring that all of them pass. Our GitHub Actions now do a linter check in frontend, backend, admin-backed, and tests folders, and run backend tests to insure backend code works. This runs every time there is a PR and nothing can be merged unless they all pass. Check the **project-setup.md** to see how to manually check your code with the linter in each respective folder.

****Note that all of the current backend routes are not only set up but tested so they can now be plugged into the frontend via the login page and they will work.**

### CODE REVIEW

File: app/backend/server.js [hunk 1]
```diff
@@ -22,19 +20,119 @@ const pool = new Pool({
...
+    })
+);
+
+app.use(passport.initialize());
+app.use(passport.session());
+passport.use(
+    new LocalStrategy({ usernameField: 'email' }, async (email, password, done) => {
+        try {
+            const userResult = await pool.query('SELECT * FROM Users WHERE email = $1 AND is_deleted = FALSE', [email]);
+            if (userResult.rowCount === 0) {
```
Comment (tannerbjorgan) [inline app/backend/server.js:46]: This should be changed to ``` if(!userResult || userResult.rowCount === 0) ```. Its proper error handling for that.

File: app/backend/server.js [hunk 2]
```diff
@@ -22,19 +20,119 @@ const pool = new Pool({
...
+// Serialize user
+passport.serializeUser((user, done) => {
+    done(null, user.user_id);
+});
+
+// Deserialize user
+passport.deserializeUser(async (id, done) => {
+    try {
+        const userResult = await pool.query('SELECT * FROM Users WHERE user_id = $1', [id]);
+        if (userResult.rowCount === 0) {
```
Comment (tannerbjorgan) [inline app/backend/server.js:75]: Same this for here as above.  ``` if(!userResult || userResult.rowCount === 0) ```

**Review by tannerbjorgan — CHANGES_REQUESTED** [review]

File: app/backend/server.js [hunk 3]
```diff
@@ -22,19 +20,119 @@ const pool = new Pool({
...
+app.use(express.json());
+app.use(express.urlencoded({ extended: false }));
+
+app.use(
+    session({
+        store: new PgSession({
+            pool,
+            tableName: 'User_Sessions',
+        }),
+        secret: 'your_secret_key',
```
[the remark this answers is not in the document]
Comment (colton456p) [reply app/backend/server.js]: Good point. I made this change.

**Review by tannerbjorgan — COMMENTED** [review]
All changes have been done and fixed. Looks good

**Review by tannerbjorgan — APPROVED** [review]

**Pull request merged**